upload/admin/controller/extension/payment/coingate.php

```diff
@@ -2,110 +2,156 @@
-    $data['footer'] = $this->load->controller('common/footer');
-
-    $this->response->setOutput($this->load->view('extension/payment/coingate', $data));
-  }
-
-  protected function validate() {
-    if (!$this->user->hasPermission('modify', 'extension/payment/coingate')) {
-      $this->error['warning'] = $this->language->get('error_permission');
+    protected function validate()
+    {
+        if (!$this->user->hasPermission('modify', 'extension/payment/coingate')) {
+            $this->error['warning'] = $this->language->get('error_permission');
+        }
+
+        if (!class_exists('CoinGate\Client')) {
+            $this->error['warning'] = $this->language->get('error_composer');
+        }
+
+        if (!$this->error) {
+            try {
+                \CoinGate\Client::setAppInfo('OpenCart', '1.0.0');
+
+                $testConnection = \CoinGate\Client::testConnection(
+                    $this->request->post['payment_coingate_api_auth_token'],
+                    $this->request->post['payment_coingate_test_mode'] == 1
+                );
+
+                if ($testConnection !== true) {
+                    $this->error['warning'] = $this->language->get('error_connection');
+                }
+            } catch (Exception $e) {
+                $this->error['warning'] = $this->language->get('error_connection');
+            }
+        }
+
+        return !$this->error;
     }
 
-    if (!class_exists('CoinGate\CoinGate')) {
-      $this->error['warning'] = $this->language->get('error_composer');
-    }
 
-    if (!$this->error) {
-      $testConnection = \CoinGate\CoinGate::testConnection(array(
-        'environment'   => $this->request->post['payment_coingate_test_mode'] == 1 ? 'sandbox' : 'live',
-        'auth_token'    => $this->request->post['payment_coingate_api_auth_token']));
+    public function install()
+    {
+        $this->load->model('extension/payment/coingate');
 
-      if ($testConnection !== true) {
-        $this->error['warning'] = $testConnection;
-      }
+        $this->model_extension_payment_coingate->install();
     }
 
-    return !$this->error;
-  }
+    public function uninstall()
+    {
+        $this->load->model('extension/payment/coingate');
 
+        $this->model_extension_payment_coingate->uninstall();
+    }
 
-	public function install() {
-    $this->load->model('extension/payment/coingate');
-    
-
-		$this->model_extension_payment_coingate->install();
-	}
+    public function getReceiveCurrencies(): array
+    {
+        $client = new CoinGate\Client();
+        $currencies = $client->getMerchantPayoutCurrencies();
 
-	public function uninstall() {
-		$this->load->model('extension/payment/coingate');
+        $currencies = array_map(function ($currency) {
+            return $currency['symbol'];
+        }, (array)$currencies);
 
-		$this->model_extension_payment_coingate->uninstall();
-	}
+        return array_merge($currencies, ['DO_NOT_CONVERT']);
```

Comment:
could it be possible to display the full name and save the symbol ?

Reply:
Yes, I changed the code as you wish.